### PR TITLE
Eliminate flashing. Only one scroll command.

### PR DESCRIPTION
**Pull request by vdbergh**
This is achieved in a maintainable way by using promises instead of events.

### CODE REVIEW

**Comment by ppigazzini** [comment]
DEV updated. I can test only with the smartphone right now.

**Comment by ppigazzini** [comment]
DEV updated.

**Comment by ppigazzini** [comment]
On the smartphone screen it seems to work fine.

**Comment by vdbergh** [comment]
Tested fine on Google-chrome, Firefox and Konqueror on Linux and Safari on iPad.

**Comment by vdbergh** [comment]
I think I finally grokked promises.

**Review by ppigazzini — APPROVED** [review]

**Comment by ppigazzini** [comment]
PROD updated as promised, thank you @vdbergh :)

File: server/fishtest/static/js/spsa.js [hunk 1]
```diff
@@ -205,9 +206,8 @@
     }, 150);
 
     //load google library
-    google.charts.load("current", {
-      packages: ["corechart"],
-      callback: function () {
+      return google.charts.load("current", {packages: ["corechart"]}).then(
+	() => {
```
Comment (dav1312) [inline server/fishtest/static/js/spsa.js:210]: Tab 👀

File: server/fishtest/static/js/spsa.js [hunk 2]
```diff
@@ -326,7 +326,6 @@
 
             redraw(false);
           });
-      },
-    });
+	});
```
Comment (dav1312) [inline server/fishtest/static/js/spsa.js:329]: Tab 👀

**Comment by ppigazzini** [comment]
@dav1312 fixed with:
https://github.com/glinscott/fishtest/commit/66a2a8f11d48813dbc90d605fc4cd6d31203a1b6
https://github.com/glinscott/fishtest/commit/1c3504ebc695a65b38a7e97934d14656c1b70a8a

**Pull request merged**